# TRIFID: determining functional isoforms

<div align="center">
  <img src="../ismb-presentation/poster/EvoCompGen_826.jpg"><br>
</div>



Video presentation and poster session prepared for the annual International Conference on Intelligent Systems for Molecular Biology (ISMB)

## COSI: EvoCompGen 826 (poster session)

**Authors**: *Fernando Pozo Ocampo, Laura Martinez Gomez and Michael Tress*

**Institution**: *Spanish National Cancer Research Centre, Spain*

**Short Abstract**: Alternative Splicing (AS) of messenger RNA can generate a wide variety of mature RNA transcripts and this expression is confirmed by experimental transcript evidence. In theory these transcripts could generate protein isoforms with diverse cellular functions. However, while peptide evidence strongly supports a main protein isoform for the vast majority of coding genes, it is not clear what proportion of these AS isoforms form stable functional proteins. In fact reliable proteomics experiments have found little evidence of alternative spliced proteins, so the number of stably folded/functional proteins produced by AS remains a mystery.

We have developed a computational method (TRIFID) for the classification of splice isoform functional importance. This machine-learning algorithm was trained on reliable peptide evidence from proteomics analyses and classifies biologically important splice isoforms with high confidence. The algorithm ranks the most significant biological splice isoforms and we show that the highest scoring alternative exons are actually under selection pressure, unlike the vast majority of alternative exons. TRIFID can predict functional isoforms for any well-annotated eukaryotic species. The method will generate valuable insights into the cellular importance of alternative splicing.

Download and visualize the poster presentation in pdf (reference and clickable links) [here](https://gitlab.com/fpozoc/ismb2020_evocompgen_826/-/raw/master/poster/EvoCompGen_826.pdf?inline=false).