---
# Basics
title: 'TRIFID: determining functional isoforms'
author:
  - name: Fernando Pozo Ocampo
    affil: 1
    orcid: '0000-0001-7688-6045'
    twitter: '@fpozoca'
    email: 'fpozoc@cnio.es'
  - name: Laura Martínez-Gómez
    affil: 1
  - name: Michael L. Tress
    affil: 1
affiliation:
  - num: 1
    address: Bioinformatics Unit, Spanish National Cancer Research Center (CNIO)
poster_width: '48in'
poster_height: '36in'
font_family: 'Archivo'
titletext_fontfamily: 'Archivo'

# Main Colour Scheme
primary_colour: '#B9BEB7' 
secondary_colour: '#332A38' 
accent_colour: '#304D6D' 

# Further Customization
# Title Box
titlebox_bordercol:	"#0b4545"	# Colour of the title box border.
title_textsize: "202px" # Poster title fontsize.
titlebox_borderwidth:	"0in"	# Width of the title box border.
title_textcol:	"#ffffff"	# Colour of title text
author_textcol:	"#ffffff"	# Colour of author text.
author_textsize:	"50pt"	# Author list font size.
affiliation_textsize:	"#ACA6A7"	# Affiliation font size.
affiliation_textcol:	"#ffffff"	# Affiliation font size.
logoleft_name:	"img/bu.png"	# Location of the image to use.
logoright_name:	"img/gencode.png"

# Poster Body
body_bgcol: "#ffffff"	 # Colour of the poster main background.
body_textsize: "60px" # Size of the main poster body text.
body_textcol:	"#000000"	# Colour of main text in the body of poster.
column_numbers:	3	# Number of columns that the poster body has.
column_margins:	"0.5in"	# Margin spacing for columns.
columnline_col:	"#d7d8db"	# Colour of the column line dividers.
columnline_width:	"2mm"	# Width of line between each column.
columnline_style:	'solid'	# Style of the column line seperator.
sectitle_textcol:	"#ffffff"	# Colour of the poster section titles.
sectitle_textsize:	"100px"	# Text size of the section titles (H1).
sectitle2_textsize:	"40pt"	# Text size of the section titles (H2).
sectitle_bgcol:	"#B9BEB7"	# Colour of the section title box.
sectitle_bordercol:	"#ffffff"	# Colour of the border around the section title box.
sectitle_borderwidth:	"2mm"	# Thicknes of the section title box border.
sectitle_boxshape:	"4mm 1mm"	# Changes the shape of the section title box.
sectitle2_textcol:	"#B9BEB7"	# Color of 2nd level header text.
reference_textsize: "20px"
link_col:	"#0b4545"	# Colour of other links within the poster.

output: 
  posterdown::posterdown_html:
    self_contained: false
    fig_caption: yes
    number_sections: false

---
<style>
p.caption {
  font-size: 0.80em;
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

# Introduction
## Background

* Alternative Splicing (AS) of messenger RNA can generate a wide variety of mature RNA transcripts and this expression is confirmed by experimental transcript evidence. However, it is not clear how many alternative transcripts will code for functional proteins. ([Tress et al. 2017](), [Blencowe et al. 2017]()).
 
* Proteomics analyses have shown that most coding genes have a single main splice isoform ([Ezkurdia et al. 2015]()).

* Human population variation data indicate that most AS transcripts are evolving neutrally ([Tress et al. 2017]()).

* Ever more splice isoforms are annotated every day, but we do not know their function.

## Objectives

<div style="text-align: left; font-size: 90%; font-weight: bold;", markdown = "1">
1. Grading what proportion of Alternative Splicing is functional.
<br><br>
2. Developing a ML based tool for predicting splice isoform functional importance.
</div>


# Methods

`TRIFID` (Tool to Reliable Identification of Functional Isoform Data) is a Random Forest based **predictor of the relative functional importance of splice isoforms**.

It has been trained on reliable peptide evidence of 497 genes from the largest tissue-based proteomics analysis to date (79 experiments) and 47 features categorized in 5 groups (*genome annotation*, *structural*, *splicing impact*, *cross-species conservation* and *RNA-seq expression*).

Code development available [here](https://gitlab.com/bu_cnio/trifid).

<div>
</div>

# Results
## SHAP feature importances and model interpretation
```{r figure1-pkgdown, out.width="100%", fig.cap='SHAP values ([Lundberg et al. 2020](https://www.nature.com/articles/s42256-019-0138-9)) added by category. Features that best distinguish functional isoforms in the training set are conservation-based. Other important features include the length difference between the alternative isoform and the longest isoform, whether or not the transcript has a CCDS ([Pruitt et al. 2009](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2704439/)), and the conservation of Pfam functional domains. This predictions also can provide clues to the influence of features on individual predictions (see example: [Fibroblast growth factor receptor 1](https://www.ensembl.org/Homo_sapiens/Gene/Summary?g=ENSG00000077782;r=8:38400215-38468834) ([TRIFID Scores](https://gitlab.com/bu_cnio/trifid#example-fibroblast-growth-factor-receptor-1-fgfr1))'}
knitr::include_graphics("img/figure1.png")
```

## Functional importance in the human genome

````{r figure2-pkgdown, out.width="100%", fig.cap='Non-redundant isoforms divided into PRINCIPAL or ALTERNATIVE according to their annotation in APPRIS ([Rodriguez et al. 2017](https://academic.oup.com/nar/article/46/D1/D213/4561658)). Most AS isoforms have TRIFID scores below 0.05. Most PRINCIPAL isoforms have predictor scores above 0.9.'}
knitr::include_graphics("img/figure2.png")
```

<div>
<br><br>
</div>

## Validating the model against an external source of information

```{r figure7-pkgdown, out.width="100%", fig.cap='It shows non-synonymous to synonymous ratios for alternative exons (exons that are exclusively present in alternative isoforms). Exons under selective pressure should have significantly lower dn/ds ratios for common than for rare allele frequencies.'}
knitr::include_graphics("img/figure3.png")
```

# Conclusions

* It discriminates functionally important isoforms with high confidence (MCC=0.89, AUPRC=0.98 over 5-folds CV of the training set).

* It predicts that a large majority of splice variants (85-90%) in the human genome are likely to not be functionally important at the protein level.

* *dn/ds* ratios from show that exons from the highest scoring of alternative transcripts are under selective pressure, while low scoring exons have little or no evidence of selection.

* The model can be succesfully exported to different genome species and genome annotation databases.