<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1">


<!--
Font-awesome icons ie github or twitter
-->
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/all.css" integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf" crossorigin="anonymous">
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/brands.css" integrity="sha384-n9+6/aSqa9lBidZMRCQHTHKJscPq6NW4pCQBiMmHdUCvPN8ZOg2zJJTkC7WIezWv" crossorigin="anonymous">

<!--
Google fonts api stuff
-->
<link href='https://fonts.googleapis.com/css?family=Open Sans' rel='stylesheet'>
<link href='https://fonts.googleapis.com/css?family=Open Sans' rel='stylesheet'>



<title>TRIFID: determining functional isoforms</title>





<style>
@page {
size: 48in 36in;
margin: 0;
padding: 0;
}
body {
margin: 0;
font-size: 51px;
width: 48in;
height: 36in;
padding: 0;
text-align: justify;
font-family: Open Sans;
}
.poster_wrap {
width: 48in;
height: 36in;
padding: 0cm;
}
.title_container {
width: 48in;
height: calc(36in * 0.15);
overflow: hidden;
background-color: #335c67;
border: 0in solid #332A38;
}
.logo_left {
float: left;
width: 10%;
height: 100%;
background-color: #335c67;
display: flex;
align-items: center;
justify-content: center;
}
.logo_right {
float: right;
width: 10%;
height: 100%;
background-color: #335c67;
display: flex;
align-items: center;
justify-content: center;
}
.poster_title {
text-align: center;
position: relative;
float: left;
width: 80%;
height: 100%;
color: #ffffff;
top: 50%;
transform: translateY(-50%);
-webkit-transform: translateY(-50%);
}
#title {
font-family: Open Sans;
}
/* unvisited link */
a:link {
color: #8d99ae;
}
.mybreak {
  break-before: column;
}
/* visited link */
a:visited {
color: #8d99ae;
}

/* mouse over link */
a:hover {
color: #8d99ae;
}

/* selected link */
a:active {
color: #8d99ae;
}
.poster_body {
-webkit-column-count: 3; /* Chrome, Safari, Opera */
-moz-column-count: 3; /* Firefox */
column-count: 3;
-webkit-column-fill: auto;
-moz-column-fill: auto;
column-fill: auto;
-webkit-column-rule-width: 2mm;
-moz-column-rule-width: 2mm;
column-rule-width: 2mm;
-webkit-column-rule-style: solid;
-moz-column-rule-style: solid;
column-rule-style: solid;
-webkit-column-rule-color: #d7d8db;
-moz-column-rule-color: #d7d8db;
column-rule-color: #d7d8db;
column-gap: 0in;
padding-left: calc(0in / 2);
padding-right: calc(0in / 2);
height: 100%;
color: #000000
background-color: #ffffff;
}
.poster_title h1 {
font-size: 160px;
margin: 0;
border: 0;
font-weight: normal;
}
.poster_body_wrap{
width: calc(48in + 0in + 0in);
height: calc(36in * 0.83);
padding-top: calc(36in * 0.005);
padding-bottom: calc(36in * 0.01);
background-color: #ffffff;
}
.poster_title h3 {
color: #ffffff;
font-size: 60px;
margin: 0;
border: 0;
font-weight: normal;
}
.poster_title h3 > sup {
  font-size: 35pt;
  margin-left: 0.02em;
}
.poster_title h5 {
color: #ffffff;
font-size: ;
margin: 0;
border: 0;
font-weight: normal;
}
img {
margin-top: 2cm;
margin-bottom: 0;
}
.section {
  padding: 0.2em;
}
.poster_body h1 {
text-align: center;
color: #ffffff;
font-size: 100px;
border: 2mm solid #ffffff;
background-color: #335c67;
border-radius: 4mm 1mm;
margin-top: 2mm;
margin-bottom: 2mm;
font-weight: normal;
}
.poster_body h2 {
color: #8d99ae;
font-size: 38pt;
padding-left: 4mm;
font-weight: normal;
}
.span {
width: 200%;
}
/* center align leaflet map,
from https://stackoverflow.com/questions/52112119/center-leaflet-in-a-rmarkdown-document */
.html-widget {
margin: auto;
position: sticky;
margin-top: 2cm;
margin-bottom: 2cm;
}
.leaflet.html-widget.html-widget-static-bound.leaflet-container.leaflet-touch.leaflet-fade-anim.leaflet-grab.leaflet-touch-drag.leaflet-touch-zoom {
position: sticky;
width: 100%;
}
pre.sourceCode.r {
background-color: #dddddd40;
border-radius: 4mm;
padding: 4mm;
width: 75%;
/* align-items: center; */
margin: auto;
padding-left: 2cm;
}
code.sourceCode.r{
background-color: transparent;
font-size: 20pt;
border-radius: 2mm;
}
.caption {
font-size: 25pt;
}
.table caption {
font-size: 25pt;
padding-bottom: 3mm;

}
code {
font-size: 1em;
font-family: monospace;
background-color: #332A3824;
color: #335c67;
padding: 1.2mm;
border-radius: 2mm;
}
.poster_title code {
font-size: 1em;
}
table {
font-size: 40px;
margin: auto;
border-top: 3px solid #666;
border-bottom: 3px solid #666;
}
table thead th {
border-bottom: 3px solid #ddd;
}
td {
padding: 8px;
}
th {
padding: 15px;
}
caption {
margin-bottom: 10px;
}
.poster_body p {
margin-right: 4mm;
margin-left: 4mm;
margin-top: 6mm;
margin-bottom: 10mm;
}
.poster_body ol {
margin-right: 4mm;
margin-left: 4mm;
}
#ul {
margin-right: 4mm;
margin-left: 4mm;
}
.references p {
font-size: 20pt;
}
.orcid img {
  width: 1em;
}
</style>
</head>
<body>


<div class="poster_wrap">
<div class="title_container">
<!-- Left Logo  -->
<div class="logo_left">
<img src= 'img/bu_cnio.png' style="width: 80%">
</div>
<!-- Poster Title -->
<div class= "poster_title">
<h1 id="title">TRIFID: determining functional isoforms</h1>
<h3 id="author"><strong>Fernando Pozo Ocampo</strong><sup>, <a class="orcid" href="https://orcid.org/0000-0001-7688-6045"><img src="https://raw.githubusercontent.com/brentthorne/posterdown/master/images/orcid.jpg"></a></sup>, Laura Martínez-Gómez<sup></sup>, Michael L. Tress<sup></sup></h3><br>
<h5 id="affiliation"><sup></sup> Bioinformatics Unit, Spanish National Cancer Research Center (CNIO)</h5>
</div>
<!-- Right Logo  -->
<div class="logo_right">
<img src=img/gencode.png style="width: 80%">
</div>
</div>

<div class='poster_body_wrap'>
<div class='poster_body'>
<style>
p.caption {
  font-size: 0.80em;
}
</style>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<div id="background" class="section level2">
<h2>Background</h2>
<p>Alternative Splicing (AS) of messenger RNA can generate a wide variety of mature RNA transcripts and this expression is confirmed by experimental transcript evidence. However, it is not clear how many alternative transcripts will code for functional proteins. (<a href="https://www.cell.com/trends/biochemical-sciences/pdf/S0968-0004(17)30071-3.pdf">Tress et al. 2017a</a>, <a href="https://www.cell.com/trends/biochemical-sciences/pdf/S0968-0004(17)30070-1.pdf">Blencowe et al. 2017</a>).</p>
<p>Proteomics analyses have shown that most coding genes have a single main splice isoform (<a href="https://pubs.acs.org/doi/10.1021/pr501286b">Ezkurdia et al. 2015</a>).</p>
<p>Human population variation data indicate that most alternative transcripts are evolving neutrally (<a href="https://www.sciencedirect.com/science/article/pii/S0968000416301189?via%3Dihub">Tress et al. 2017</a>).</p>
<p>Ever more splice isoforms are annotated every day, but we do not know their function.</p>
</div>
<div id="objectives" class="section level2">
<h2>Objectives</h2>
<div style="text-align: left; font-size: 110%; font-weight: bold;", markdown = "1">
Grading what proportion of AS is functional.
<br><br>
Developing a machine learning based tool for predicting splice isoform functional importance.
</div>
</div>
</div>
<div id="methods" class="section level1">
<h1>Methods</h1>
<p><code>TRIFID</code> (Tool to Reliable Identification of Functional Isoform Data) is a Random Forest based <strong>predictor of the relative functional importance of splice isoforms</strong>.</p>
<p>It has been trained on reliable peptide evidence from 497 genes from the largest tissue-based proteomics analysis to date (79 experiments) and 47 features categorized in 5 groups (<em>genome annotation</em>, <em>structural</em>, <em>splicing impact</em>, <em>cross-species conservation</em> and <em>RNA-seq expression</em>).</p>
<p><a href="https://gitlab.com/bu_cnio/trifid">Development code</a> and <a href="https://gitlab.com/fpozoc/ismb2020_evocompgen_826">poster presentation</a> repositories.</p>
<div>

</div>
</div>
<div id="results" class="section level1">
<h1>Results</h1>
<div id="shap-feature-importances-and-model-interpretation" class="section level2">
<h2>SHAP feature importances and model interpretation</h2>
<div class="figure"><span id="fig:figure1-pkgdown"></span>
<img src="img/figure1.png" alt="SHAP values ([Lundberg et al. 2020](https://www.nature.com/articles/s42256-019-0138-9)) added by category. Features that best distinguish functional isoforms in the training set are conservation-based. Other important features include the length difference between the alternative isoform and the longest isoform, whether or not the transcript has a CCDS ([Pruitt et al. 2009](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2704439/)), and the conservation of Pfam functional domains. The SHAP scores can also provide clues to the influence of features on individual predictions (see example: Fibroblast growth factor receptor 1 ([TRIFID Scores](https://gitlab.com/bu_cnio/trifid#example-fibroblast-growth-factor-receptor-1-fgfr1))" width="90%" />
<p class="caption">
Figure 1: SHAP values (<a href="https://www.nature.com/articles/s42256-019-0138-9">Lundberg et al. 2020</a>) added by category. Features that best distinguish functional isoforms in the training set are conservation-based. Other important features include the length difference between the alternative isoform and the longest isoform, whether or not the transcript has a CCDS (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2704439/">Pruitt et al. 2009</a>), and the conservation of Pfam functional domains. The SHAP scores can also provide clues to the influence of features on individual predictions (see example: Fibroblast growth factor receptor 1 (<a href="https://gitlab.com/bu_cnio/trifid#example-fibroblast-growth-factor-receptor-1-fgfr1">TRIFID Scores</a>)
</p>
</div>
</div>
<div id="functional-importance-in-the-human-genome" class="section level2">
<h2>Functional importance in the human genome</h2>
<div class="figure"><span id="fig:figure2-pkgdown"></span>
<img src="img/figure2.png" alt="Non-redundant isoforms divided into PRINCIPAL or ALTERNATIVE according to their annotation in APPRIS ([Rodriguez et al. 2017](https://academic.oup.com/nar/article/46/D1/D213/4561658)). Most ALTERNATIVE isoforms have TRIFID scores below 0.05. Most PRINCIPAL isoforms have predictor scores above 0.9." width="100%" />
<p class="caption">
Figure 2: Non-redundant isoforms divided into PRINCIPAL or ALTERNATIVE according to their annotation in APPRIS (<a href="https://academic.oup.com/nar/article/46/D1/D213/4561658">Rodriguez et al. 2017</a>). Most ALTERNATIVE isoforms have TRIFID scores below 0.05. Most PRINCIPAL isoforms have predictor scores above 0.9.
</p>
</div>
<div style="text-align: center; font-size: 65%; font-style: italic;", markdown = "1">
<br>
<br>
<br>
This work has been funded by the US National Institutes of Health grant 2 U41 HG007234.
<br>
</div>
</div>
<div id="validating-the-model-against-an-external-source-of-information" class="section level2">
<h2>Validating the model against an external source of information</h2>
<div class="figure"><span id="fig:figure3-pkgdown"></span>
<img src="img/figure3.png" alt="Non-synonymous to synonymous (NS/Syn) ratios for alternative exons (exons that are exclusively present in alternative isoforms). Exons under selective pressure should have significantly lower NS/Syn ratios for common than for rare allele frequencies. The vast majority of alternative exons are not under selective pressure." width="90%" />
<p class="caption">
Figure 3: Non-synonymous to synonymous (NS/Syn) ratios for alternative exons (exons that are exclusively present in alternative isoforms). Exons under selective pressure should have significantly lower NS/Syn ratios for common than for rare allele frequencies. The vast majority of alternative exons are not under selective pressure.
</p>
</div>
</div>
</div>
<div id="conclusions" class="section level1">
<h1>Conclusions</h1>
<p><code>TRIFID</code> discriminates functionally important isoforms with high confidence (MCC=0.89, AUPRC=0.98 over 5-folds CV of the training set). Thanks to TRIFID, we can now list the most biologically relevant alternative isoforms. Our model can be successfully exported to different genome species and genome annotation databases.</p>
<p><code>TRIFID</code> predicts that a large majority of splice variants (85-90%) in the human genome are likely to not be functionally important at the protein level.</p>
<p>NS/Syn ratios show that exons from the highest scoring of alternative transcripts are under selective pressure, while low scoring exons have little or no evidence of selection.</p>
<!-- # Acknowledgements

. -->
</div>
</div>
</div>

</div>



</body>
</html>
